src/model.rs

```diff
@@ -24,67 +25,49 @@ use crate::{
     utils::{conf::SharedConf, crypto::SharedBlstCrypto},
 };
 
-#[derive(Default, Clone, Eq, PartialEq, Hash, Encode, Decode)]
-pub struct TxHash(pub Vec<u8>);
+#[derive(
+    Debug, Display, Default, Clone, Eq, PartialEq, Hash, Encode, Decode, Deserialize, Serialize,
+)]
+pub struct TxHash(pub String);
 
-impl TxHash {
-    pub fn new(s: &str) -> TxHash {
-        TxHash(s.into())
+impl Type<Postgres> for TxHash {
+    fn type_info() -> sqlx::postgres::PgTypeInfo {
+        <String as Type<Postgres>>::type_info()
     }
 }
-
-impl Display for TxHash {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", hex::encode(&self.0))
+impl<'q> sqlx::Encode<'q, sqlx::Postgres> for TxHash {
+    fn encode_by_ref(
+        &self,
+        buf: &mut sqlx::postgres::PgArgumentBuffer,
+    ) -> std::result::Result<
+        sqlx::encode::IsNull,
+        std::boxed::Box<(dyn std::error::Error + std::marker::Send + std::marker::Sync + 'static)>,
+    > {
+        <String as sqlx::Encode<sqlx::Postgres>>::encode_by_ref(&self.0, buf)
```

Comment:
I will have a problem with that soon: I move TxHash ton an extern crate "sdk", thus I won't be able to impl this sqlx :grimacing:

Reply:
Yeah, thought about that :smile: ...
That's what needed if we want to keep TxHash as a type... Otherwise we could just mark it as String...

Reply:
We definitively wnat to keep it as a type, I'll find a workaround

I'm thinking about a TxHashDb that can easily convert from/to TxHash, I'll see what I can do once we merged this PR